src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/GenericModelBinder.cs

```diff
@@ -34,27 +40,26 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
                 !collectionBinder.CanCreateInstance(bindingContext.ModelType))
             {
                 // Able to resolve a binder type but need a new model instance and that binder cannot create it.
-                return ModelBindingResult.NoResultAsync;
+                return Internal.TaskCache.CompletedTask;
             }
 
             return BindModelCoreAsync(bindingContext, binder);
         }
 
-        private async Task<ModelBindingResult> BindModelCoreAsync(ModelBindingContext bindingContext, IModelBinder binder)
+        private async Task BindModelCoreAsync(IModelBindingContext bindingContext, IModelBinder binder)
         {
             Debug.Assert(binder != null);
 
-            var result = await binder.BindModelAsync(bindingContext);
-            var modelBindingResult = result != ModelBindingResult.NoResult ?
-                result :
-                ModelBindingResult.Failed(bindingContext.ModelName);
-
-            // Were able to resolve a binder type.
-            // Always tell the model binding system to skip other model binders.
-            return modelBindingResult;
+            await binder.BindModelAsync(bindingContext);
+            if (bindingContext.Result == null)
+            {
+                // Were able to resolve a binder type.
+                // Always tell the model binding system to skip other model binders.
```

Comment:
Not your change but something has been lost here. The code at one point matched the comment -- always returning something _other_ than `NoResult`. Likely the comment needs to change.